discovery/gossiper.go

```diff
@@ -753,7 +753,11 @@ func (d *AuthenticatedGossiper) stop() {
 	log.Debug("Authenticated Gossiper is stopping")
 	defer log.Debug("Authenticated Gossiper stopped")
 
-	d.blockEpochs.Cancel()
+	// `blockEpochs` is only initialized in the start routine so we make
+	// sure we don't panic here.
```

Comment:
nit:  "we don't panic here in the case where Stop is called when Start does not complete"